src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -5019,6 +5036,36 @@ void ObjectMgr::LoadQuests()
         }
     }
 
+    // Disallow any breadcrumb loops and inform quests of their breadcrumbs
+    for (auto& questPair : _questTemplates)
+    {
+        // skip post-loading checks for disabled quests
+        if (DisableMgr::IsDisabledFor(DISABLE_TYPE_QUEST, questPair.first, nullptr))
+            continue;
+
+        Quest* qinfo = &questPair.second;
+        uint32   qid = qinfo->GetQuestId();
+        uint32 breadcrumbForQuestId = std::abs(qinfo->_breadcrumbForQuestId);
+        std::set<uint32> questSet;
+
+        while(breadcrumbForQuestId)
+        {
+            //a previously visited quest was found as a breadcrumb quest
+            //breadcrumb loop found!
+            if (!questSet.insert(qinfo->_id).second)
+            {
+                TC_LOG_FATAL("sql.sql", "Breadcrumb quests %u and %u are in a loop", qid, breadcrumbForQuestId);
+                exit(1);
```

Comment:
please just gracefully handle the case and log only a TC_LOG_ERROR, there's not need to shutdown worldserver for just a quest

Reply:
No, this one will infinite recursive loop the server later if there's a loop.
I'm of the opinion that all these checks should hard fault and tell the devs to fix their damn tables.

Reply:
The point is that we are the devs, not the users using TC. If we push a faulty SQL update, we will break all users' TC for no apparent reason as we can just force _breadcrumbForQuestId to 0 when a loop is found.

Reply:
Let's hope none will add bad data then, but please at least fix the one above about quest not found

Reply:
As you wanted; both cases handled "gracefully".
As to TC devs breaking stuff, that happens all the time and it's what the snail pace review process is for.
Strict error warnings are only a benefit and would immediately eliminate those kinds of errors as the only solution is for devs to fix their damn tables (can't accidentally push a faulty sql when the server won't even start), especially as the wiki documentation is woefully out of date on many issues.